web-console/src/views/sql-view.tsx

```diff
@@ -62,25 +64,33 @@ export class SqlView extends React.Component<SqlViewProps, SqlViewState> {
       explainDialogOpen: false,
       loadingExplain: false,
       explainResult: null,
-      explainError: null
+      explainError: null,
+      queryElapsed: 0
     };
   }
 
   componentDidMount(): void {
     this.sqlQueryManager = new QueryManager({
       processQuery: async (query: string) => {
+        const startTime = new Date();
         if (query.trim().startsWith('{')) {
           // Secret way to issue a native JSON "rune" query
           const runeQuery = Hjson.parse(query);
-          return decodeRune(runeQuery, await queryDruidRune(runeQuery));
+          const result = await queryDruidRune(runeQuery);
+          this.setState({
```

Comment:
do not call setState in `processQuery` you are making this function not pure. pass it though to `onStateChange`